synchronization/src/main/java/de/cyface/synchronization/HttpConnection.java

```diff
@@ -265,9 +274,11 @@ private String addPart(final @NonNull String key, final @NonNull String value, f
      * @throws ResponseParsingException If the system fails in handling the HTTP response.
      * @throws UnauthorizedException If the credentials for the cyface server are wrong.
      * @throws SynchronisationException If a connection error occurred while reading the response code.
+     * @throws NetworkErrorException when the connection's input or error stream was null
      */
-    private HttpResponse readResponse(final @NonNull HttpURLConnection con) throws ResponseParsingException,
-            DataTransmissionException, UnauthorizedException, SynchronisationException, BadRequestException {
+    private HttpResponse readResponse(final @NonNull HttpURLConnection con)
```

Comment:
Didn't we say some time we would move the Annotation in front of the final?

Reply:
We did. I didnt add this line now but just removed or added exceptions so in this special case I did not clean up all old lines such as this. But I'm doing this very often these days - changing the order correctly.